src/radical/entk/pipeline.py

```diff
@@ -406,9 +406,29 @@ def _validate(self):
 
             raise EnTKMissingError(obj=self._uid, missing_attribute='stages')
 
+        annotated_outputs = {}
         for stage in self._stages:
             stage._validate()  # pylint: disable=W0212
 
+            # validate provided annotations
+            for task in stage.tasks:
+
+                if not task.annotations:
+                    continue
+
+                annotated_outputs[task.uid] = task.annotations.outputs
+                for t_input in task.annotations.inputs:
+                    if ':' not in t_input:
+                        continue
+                    # ensure that task inputs are annotated as outputs
+                    # for corresponding tasks it depends on
+                    t_dep_uid, t_dep_output = t_input.split(':')
```

Comment:
Catch only the first part

Reply:
@mturilli I've extended this coverage, ready for the next round review/merge